bitblas/ops/operator.py

```diff
@@ -50,26 +51,36 @@ def __init__(self, config: OperatorConfig):
         assert self.is_valid_config(config), (f"Invalid config for {self.__class__.__name__}: "
                                               f"{config}")
         self.config = config
+        self.kernel_name = ""
 
     @abstractmethod
     def is_valid_config(self, config: OperatorConfig):
         pass
 
     @abstractmethod
-    def generate(self, hint: Hint = None) -> str:
+    def _generate(self, hint: Hint = None) -> str:
```

Comment:
it’s worth considering if we should add a leading underline before `generate`?

In python function, `_` begin method might represent the function is `protected`.

Reply:
A new function "generate_valid" will be the new interface for kernel name generate, with the call to "_generate" first and check the validity of the generated kernel name.
The protected "_generate" can be overwritten by sub class for real generation process.